src/pages/Agents/SubmittableRuns.vue

```diff
@@ -93,54 +100,69 @@ export default {
       )
     },
     title() {
-      let title = ''
-      if (this.tab == 'submittable') {
-        title =
-          this.loading > 0
-            ? 'Submittable runs'
-            : `${this.submittableRuns?.length || 0} submittable runs`
-      }
+      return this.tabProperties[this.tab].title
+    },
+    titleIcon() {
+      return this.tabProperties[this.tab].icon
+    },
+    titleIconColor() {
+      if (this.loading) return 'grey'
 
-      if (this.tab == 'late') {
-        title =
-          this.loading || this.isClearingLateRuns
-            ? 'Late runs'
-            : `${this.lateRuns?.length +
-                (this.lateRuns?.length >= 1000 ? '+' : '') || 0} late runs`
-      }
+      return this.tabProperties[this.tab].icon_color
+    },
+    systemBarColor() {
+      if (this.loading) return 'secondaryGray'
 
-      return title
+      return this.lateRuns?.length > 0 ? 'deepRed' : 'Success'
     },
-    titleIcon() {
-      let icon = ''
+    tabProperties() {
+      var submittableCount = this.submittableRuns?.length || 0
+      var lateCount = this.lateRuns?.length || 0
 
-      if (this.tab == 'submittable') {
-        icon = 'access_time'
+      return {
+        [this.tabs.submittable]: {
+          title: `${
+            submittableCount > 999 ? '1,000+' : submittableCount
+          } submittable runs`,
+          icon: 'access_time',
+          icon_color: 'primary'
+        },
+        [this.tabs.late]: {
+          title: `${lateCount > 999 ? '1,000+' : lateCount} late runs`,
+          icon: 'timelapse',
+          icon_color: lateCount > 0 ? 'deepRed' : 'Success'
+        }
       }
+    },
+    submittableTabTitle() {
+      var submittableCount = this.submittableRuns?.length || 0
 
-      if (this.tab == 'late') {
-        icon = 'timelapse'
+      if (this.tab == this.tabs.submittable || submittableCount == 0) {
+        return 'Submittable'
       }
-      return icon
+
+      return `(${
+        submittableCount > 999 ? '1,000+' : submittableCount
+      }) Submittable`
     },
-    titleIconColor() {
-      return this.loading
-        ? 'grey'
-        : this.tab == 'submittable'
-        ? 'primary'
-        : this.lateRuns?.length > 0
-        ? 'deepRed'
-        : 'Success'
+    lateTabTitle() {
+      var lateCount = this.lateRuns?.length || 0
+
+      if (this.tab == this.tabs.late || lateCount == 0) {
+        return 'Late'
+      }
+
+      return `(${lateCount > 999 ? '1,000+' : lateCount}) Late`
     }
   },
   watch: {
     agent(val) {
-      if (!val) return
-      if (val.lateRuns?.length > 0) {
-        this.tab = 'late'
-      }
-      if (val.lateRuns?.length <= 0) {
-        this.tab = 'submittable'
+      if (!val || this.userSetTab) return
+
+      if (this.tab == this.tabs.submittable && this.lateRuns?.length > 0) {
+        this.tab = this.tabs.late
+      } else if (this.tab == this.tabs.late && this.lateRuns?.length == 0) {
+        this.tab = this.tabs.submittable
```

Comment:
quick note here, I was frustrated that if there are runs in the "late" tab, even when I explicitly go back to the "submittable" tab the system would continue to kick me back to "late" on a regular interval. I change this behavior to only programmatically switch the tab if the user hasn't explicitly changed tabs (same as UpcomingRunsTile)